### PR TITLE
Rust 1.73.0 update

### PR DESCRIPTION
**Pull request by Anton-4**

### CODE REVIEW

File: crates/glue/src/RustGlue.roc [hunk 1]
```diff
@@ -2063,9 +2063,9 @@ fileHeader =
     #![allow(clippy::missing_safety_doc)]
     #![allow(clippy::let_and_return)]
     #![allow(clippy::missing_safety_doc)]
-    #![allow(clippy::redundant_static_lifetimes)]
```
Comment (Anton-4) [inline crates/glue/src/RustGlue.roc:2066]: This line was listed twice.

File: crates/glue/src/RustGlue.roc [hunk 2]
```diff
@@ -2063,9 +2063,9 @@ fileHeader =
...
     #![allow(clippy::needless_borrow)]
     #![allow(clippy::clone_on_copy)]
+    #![allow(clippy::incorrect_partial_ord_impl_on_ord_type)]
```
Comment (Anton-4) [inline crates/glue/src/RustGlue.roc:2068]: Clippy wanted to replace a function like below with some single line default, but I assume we need the custom code?
```
    fn cmp(&self, other: &Self) -> core::cmp::Ordering {
        match self.discriminant().cmp(&other.discriminant()) {
            core::cmp::Ordering::Equal => {}
            not_eq => return not_eq,
        }

        unsafe {
            match self.discriminant() {
                discriminant_RocType::Bool => core::cmp::Ordering::Equal,
                discriminant_RocType::EmptyTagUnion => core::cmp::Ordering::Equal,
                discriminant_RocType::Function => self.Function.cmp(&other.Function),
                discriminant_RocType::Num => self.Num.cmp(&other.Num),
                discriminant_RocType::RecursivePointer => {
                    self.RecursivePointer.cmp(&other.RecursivePointer)
                }
                discriminant_RocType::RocBox => self.RocBox.cmp(&other.RocBox),
                discriminant_RocType::RocDict => self.RocDict.cmp(&other.RocDict),
                discriminant_RocType::RocList => self.RocList.cmp(&other.RocList),
                discriminant_RocType::RocResult => self.RocResult.cmp(&other.RocResult),
                discriminant_RocType::RocSet => self.RocSet.cmp(&other.RocSet),
                discriminant_RocType::RocStr => core::cmp::Ordering::Equal,
                discriminant_RocType::Struct => self.Struct.cmp(&other.Struct),
                discriminant_RocType::TagUnion => self.TagUnion.cmp(&other.TagUnion),
                discriminant_RocType::TagUnionPayload => {
                    self.TagUnionPayload.cmp(&other.TagUnionPayload)
                }
                discriminant_RocType::Unit => core::cmp::Ordering::Equal,
            }
        }
    }
}
```

Comment (rtfeldman) [reply crates/glue/src/RustGlue.roc:2068]: Yeah let's leave it as-is!

**Review by rtfeldman — APPROVED** [review]
Thanks @Anton-4!

**Pull request merged**